### DISCUSSION TITLE
Dynamic prefix to ```env```

### DISCUSSION THREAD

**Discussion by codectl** [pydantic/pydantic#5027] [Question]
Suppose I have:

```python
class Settings(BaseSettings):
    TYPE: str
```

How can map a dynamic env variable like ```X_TYPE=x``` or ```Y_TYPE=y``` to ```TYPE```?

I tried parsing values with ```root_validator``` but I don't get the raw values, even with ```pre=True```.

**Comment by samuelcolvin** [comment]
No easy way, probably best to do your own processing of environment variables if they're that dynamic.
